### PR TITLE
Create date range widget edit and preview for analytical framework

### LINKED ISSUE #774: Analytical Statement: Create date range widget edit and preview

### PR DESCRIPTION
**Pull request by samshara**
fixes https://github.com/the-deep/server/issues/774

### CODE REVIEW

File: src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import WidgetWrapper from '../../Widget';
+
+export interface Props <N extends string>{
+    title: string | undefined;
+    className?: string;
+
+    name: N,
+    value: string | null | undefined,
+    onChange: (value: string | undefined, name: N) => void,
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx]: The value should not be 'string'

File: src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            title={title}
+            actions={actions}
+        >
+            <TextInput // TODO: use DateRangeInput when available
+                name={name}
+                onChange={onChange}
+                value={value}
+                readOnly={readOnly}
+                disabled={disabled}
+            />
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx]: We need to add two different TextInputs

File: src/newViews/AnalyticalFramework/types.ts [hunk 3]
```diff
@@ -111,6 +111,10 @@ interface Matrix2Data extends BaseData<string[]> {
     rows: Matrix2dRows[];
     columns: Matrix2dColumns[];
 }
+interface DateRangeData extends BaseData<string> {
+    to?: string;
+    from?: string;
+}
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/types.ts]: We may not need this interface

File: src/newViews/AnalyticalFramework/types.ts [hunk 4]
```diff
@@ -153,9 +157,9 @@ interface TimeRangeWidget extends BaseWidget {
     type: 'time-range';
     data: BaseData<string>;
 }
-interface DateRangeWidget extends BaseWidget {
+export interface DateRangeWidget extends BaseWidget {
     type: 'date-range';
-    data: BaseData<string>;
+    data: DateRangeData;
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/types.ts]:
```suggestion
    data: BaseData<{ to: string, from: string }>;
```

File: src/newViews/AnalyticalFramework/WidgetEditor/DateRangeWidgetForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                error={error?.fields?.from}
+            />
+            <DateInput
+                className={styles.input}
+                label="To"
+                name="to"
+                value={value?.to}
+                onChange={onFieldChange}
+                error={error?.fields?.to}
+            />
```
Comment (tnagorra) [inline src/newViews/AnalyticalFramework/WidgetEditor/DateRangeWidgetForm/index.tsx]: We don't save `from` and `to` value here
It's saved on WidgetPreview element.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            <DateInput
+                name={name}
+                onChange={handleFromChange}
+                value={value?.from}
+                readOnly={readOnly}
+                disabled={disabled}
+            />
+        </WidgetWrapper>
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx]: Add option to swap dates as we have on the previous widget.

Comment (AdityaKhatri) [reply src/newViews/AnalyticalFramework/WidgetPreview/DateRangeWidgetInput/index.tsx]: Also, I've rebased, fixed the styling, and pushed.

**Pull request merged**